vgaidarji/github-android-app
# Provide GitHub user name as parameter instead of hardcoding it

### What has been done
- Provided GitHub user name as parameter instead of hardcoding it. Now the user name can be configured in `local.properties`. Also, used Dagger assisted injection to provide some parameters at creation time rather than via dependency graph automatic constructor injection. https://dagger.dev/dev-guide/assisted-injection.html
- Added separate screen to type in the GitHub user name to allow for arbitrary user GitHub profile information fetching

### How to test
- Use different user names in `local.properties` and make sure they are propagated correctly to the app
- Use different user names in app and make sure app loads information correctly for them
- Check the app showcase gif in `README` (check out to PR source branch on GitHub)
		<img src="https://github.com/user-attachments/assets/5f320981-462b-4968-8699-18cfc8207277" width="280">
